[Plugin] NotesSearcher

vishnyanetchereshnya opened the pull request:
Allows you to open modal with all of your notes and search throught them by UserID, Note text and Global/Username if user is cached
![Discord_S1PuI7Rhym](https://github.com/Vendicated/Vencord/assets/151846235/06e5746c-6447-4ea1-8bcb-d1eecc68d472)

Nuckyz committed:
Clean-up related additions to mangled exports

Nuckyz committed:
Reporter: Test mapMangledModule

Vendicated committed:
Plugin Page: add indicator for excluded plugins

vishnyanetchereshnya committed:
[Plugin] NotesSearcher

Allows you to open modal with all of your notes and search throught them by UserID, Note text and Global/Username if user is cached

vishnyanetchereshnya committed:
Update index.tsx

Scyye commented:
Why not just rename the pr?

vishnyanetchereshnya committed:
simple-header

vishnyanetchereshnya commented:
> Why not just rename the pr?

feels like creating new pr is more correct ¯\_(ツ)_/¯

MrDiamondDog commented on src/plugins/notesSearcher/components/NotesDataModal.tsx:
```diff
@@ -0,0 +1,514 @@
+        <div
+            className={cl("user")}
+            onContextMenu={event => {
+                ContextMenuApi.openContextMenu(event, () =>
```
Also put this in another component

MrDiamondDog commented on src/plugins/notesSearcher/data.ts:
```diff
@@ -0,0 +1,101 @@
+    }
+};
+
+const states: {
+    setRunning?: ReturnType<typeof useState<any>>[1];
+    setCacheStatus?: ReturnType<typeof useState<any>>[1],
+} = {};
```
Make this a type so you don't have to write it out everytime

MrDiamondDog reviewed (commented):
This plugin is pretty messy, especially the `NotesDataModal.tsx` file. I would split a lot of this file into different files

vishnyanetchereshnya committed:
refactoring

vishnyanetchereshnya commented on src/plugins/notesSearcher/components/NotesDataModal.tsx:
```diff
@@ -0,0 +1,514 @@
+                </Popout>
+                <ModalCloseButton onClick={close} />
+            </ModalHeader>
+            {
```
> Why is this wrapped in {}

cos it's react?

MrDiamondDog replied:
you don't need to wrap it if there isn't any js inside it, it's just the react component so you don't need to use {}

vishnyanetchereshnya replied:
`close` is js func, it's actually `() => close()`

MrDiamondDog replied:
that's not the part of the code im talking about

You have a big `div` wrapped with `{ }` for no reason, since there isn't any js used directly in until inside the `div` where you already use `{ }`

vishnyanetchereshnya replied:
lol, haha, didn't realize what you were talking about
yeah, you right, will remove

vishnyanetchereshnya commented:
> This plugin is pretty messy, especially the `NotesDataModal.tsx` file. I would split a lot of this file into different files

👍

dolfies commented on src/plugins/notesSearcher/data.ts:
```diff
@@ -0,0 +1,103 @@
+
+    onlyMissing || usersCache.clear();
+
+    for (const userId of NotesMap.keys()) {
```
You can use the Gateway to attempt to cache up to 100 users at once (see how ImplicitRelationships does it), and then fetch the unsuccessful ones (ones that you do not have a mutual with) like this. Sending up to 1,500 requests to fetch users isn't ideal. Also, the client itself also has a user cache (that you don't appear to be using here?)

vishnyanetchereshnya replied:
👍

vishnyanetchereshnya committed:
new way of users caching & receive notes in READY

Vendicated commented:
thanks for your contribution!

without having actually read the code, this plugin seems somewhat overengineered. so much code and 300 lines of css for what's basically just a simple request + array map render

try to reduce the amount of code, prefer using discord components whereever possible

also like dolfies said, the user fetching stuff should not be exposed to the user at all. and doing it via RestAPI is api abuse, prefer using the gateway. Vencord might add a util soon that allows you to do this more easily

vishnyanetchereshnya committed:
Merge branch 'dev' into NotesSearcher

vishnyanetchereshnya committed:
Delete src/api/UserSettingDefinitions.ts

vishnyanetchereshnya committed:
Delete src/plugins/_api/userSettingsDefinitions.ts

vishnyanetchereshnya commented:
> thanks for your contribution!
> 
> without having actually read the code, this plugin seems somewhat overengineered. so much code and 300 lines of css for what's basically just a simple request + array map render seems unreasonable
> 
> try to reduce the amount of code, prefer using discord components wherever possible
> 
> also like dolfies said, the user fetching stuff should not be exposed to the user at all. and doing it via RestAPI is api abuse, prefer using the gateway. Vencord might add a util soon that allows you to do this more easily

I fully removed RestAPI from users caching and replaced it by gateway but I'm still using first one (`UserUtils.getUser`) if didn't have user in cache
btw have an idea to remove cache menu at all and just cache all users on startup + remove `UserUtils.getUser`, if user anyway didn't cached then I'll just render `Unable to load`. in addition it will reduce amount of code/css

about overengineered I'm agree with you, just love to create functionality things. and yea, I'm trying to use Discord components where possible

vishnyanetchereshnya committed:
bugfix

vishnyanetchereshnya committed:
console.log remove

dolfies commented on src/plugins/notesSearcher/index.tsx:
```diff
@@ -9,34 +9,95 @@ import "./styles.css";
+            },
+        },
+        {
+            find: ".REQUEST_GUILD_MEMBERS",
+            replacement: {
+                match: /\.send\(8,{(?!nonce)/,
+                replace: "$&nonce:arguments[1].nonce,",
+            },
+        },
+        {
+            find: "GUILD_MEMBERS_REQUEST:",
+            replacement: {
+                match: /presences:!!(\i)\.presences(?!,nonce)/,
+                replace: "$&,nonce:$1.nonce",
+            },
+        },
+        {
+            find: ".not_found",
+            replacement: {
+                match: /notFound:(\i)\.not_found(?!,nonce)/,
+                replace: "$&,nonce:$1.nonce",
+            },
+        },
```
These patches are gonna conflict with ImplicitRelationships. Maybe they should be moved to an API or something? Not sure

vishnyanetchereshnya replied:
> These patches are gonna conflict with ImplicitRelationships. Maybe they should be moved to an API or something? Not sure

I added `(?!,nonce)` to match regex so it's should be fine

Scyye replied:
im not an expert on patches, but i dont think thats gonna work.
have you tested it?

vishnyanetchereshnya replied:
> im not an expert on patches, but i dont think thats gonna work. have you tested it?

yup, `?!` in the start of capture group means match only if this one not found

vishnyanetchereshnya committed:
cache menu remove

cache menu removed -> users will be cached once modal open (try to fetch users from mutual guilds if not then by RestAPI with 1 sec delay (won't cause any api timeouts))
moved much part of css from styles.css into elements `style` prop

vishnyanetchereshnya committed:
Merge branch 'dev' into NotesSearcher

vishnyanetchereshnya committed:
Delete src/plugins/notesSearcher/components/CachePopout.tsx

vishnyanetchereshnya commented:
@Vendicated the way of caching of users is refactored and .css reduced, hope now fine about it

vishnyanetchereshnya committed:
Delete src/plugins/notesSearcher/noteStore.d.ts

vishnyanetchereshnya committed:
Update index.tsx

vishnyanetchereshnya committed:
Merge branch 'dev' into NotesSearcher

vishnyanetchereshnya committed:
header fix

dolfies reviewed (commented):
looks great!

vishnyanetchereshnya commented on src/plugins/notesSearcher/components/NotesDataRow.tsx:
```diff
@@ -0,0 +1,248 @@
+        id: userId,
+        globalName: "Loading...",
+        username: "Loading...",
+        avatar: "https://discord.com/assets/0048cbfdd0b3ef186d22.png",
```
> better than hardcoding the asset imo

even didn't know about that, thanks

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/components/NotesDataRow.tsx

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/data.ts

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya commented on src/plugins/notesSearcher/data.ts:
```diff
@@ -0,0 +1,153 @@
+    });
+};
+
+export const usersCache: t.UsersCache = new Map();
```
> why do you need your own user cache instead of using discord's store?

I think it's better to have map of users who actually have note, rather than getting them from discord store with a bunch of users, it isn't?

dolfies replied:
honestly i think it's ptobsbly fine, up to @Vendicated tho

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/components/NotesDataModal.tsx

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/components/NotesDataModal.tsx

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya commented on src/plugins/notesSearcher/components/NotesDataRow.tsx:
```diff
@@ -0,0 +1,248 @@
+                        }
+                        <Menu.MenuItem
+                            id={cl("copy-user-notes")}
+                            label="Copy Notes"
```
> I feel like this is better if it's singular instead of plural, cuz that's what the client uses elsewhere, this is done in a couple other places as well

yea, I think that's better, I'm not a native speaker so I get confused at times like this, haha

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/components/NotesDataRow.tsx

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/components/NotesDataRow.tsx

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya committed:
Update src/plugins/notesSearcher/components/NotesDataButton.tsx

Co-authored-by: dolfies <jeyalfie47@gmail.com>

vishnyanetchereshnya committed:
Merge branch 'dev' into NotesSearcher

vishnyanetchereshnya committed:
apply css only only if plugin enabled

vishnyanetchereshnya committed:
lint fix

vishnyanetchereshnya committed:
bugfix

vishnyanetchereshnya committed:
Merge branch 'dev' into NotesSearcher
